packages/parser-babel-config/src/babel-config.ts

```diff
@@ -0,0 +1,109 @@
+import * as path from 'path';
+import * as ajv from 'ajv';
+
+import { ScanEnd, FetchEnd, Parser } from 'sonarwhal/dist/src/lib/types';
+import { Sonarwhal } from 'sonarwhal';
+import { loadJSONFile } from 'sonarwhal/dist/src/lib/utils/misc';
+
+import { BabelConfig, BabelConfigInvalid, BabelConfigParsed, BabelConfigInvalidSchema } from './BabelConfigParse';
+
+export default class BabelConfigParser extends Parser {
+    private configFound: boolean = false;
+    private schema: any;
+    private validator: ajv.Ajv;
+
+    public constructor(sonarwhal: Sonarwhal) {
+        super(sonarwhal);
+        this.schema = loadJSONFile(path.join(__dirname, 'schema', 'babelConfigSchema.json'));
+
+        /*
+         * If we want to use the ajv types in TypeScript, we need to import
+         * ajv in a lowsercase variable 'ajv', otherwise, we can't use types
+         * like `ajv.Ajv'.
+         */
+        this.validator = new ajv({ // eslint-disable-line new-cap
+            $data: true,
+            allErrors: true,
+            schemaId: 'id',
+            verbose: true
+        });
+
+        this.validator.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'))
+
+        /**
+         * .babelrc => type: 'unknown' ('file-type' module doesn't support the type of 'json').
+         * package.json => type: 'json' (file type from extention).
+         */
+        sonarwhal.on('fetch::end::json', this.parseBabelConfig.bind(this));
+        sonarwhal.on('fetch::end::unknown', this.parseBabelConfig.bind(this));
```

Comment:
Are we sure we need the event `unknown`? In the TypeScript one @sarvaje didn't need it.

Reply:
@molant Confirmed, the event is `fetch::end::unknown` for `.babelrc`. In @sarvaje 's case, the config file always has `.json` extension so the media type could read directly.